src/verify-and-receive.ts

```diff
@@ -26,9 +27,20 @@ export async function verifyAndReceive(
     );
   }
 
-  return state.eventHandler.receive({
-    id: event.id,
-    name: event.name,
-    payload: JSON.parse(event.payload),
-  });
+  try {
+    return state.eventHandler.receive({
+      id: event.id,
+      name: event.name,
+      payload: JSON.parse(event.payload),
+    });
+  } catch (error: any) {
+    error.message = "Invalid JSON";
+    error.status = 400;
+    let tmpStackTraceLimit = Error.stackTraceLimit;
```

Comment:
Can you explain this change? Adding comments to the code is useful for these sorts of things

Reply:
Reverted this because on second thoughts i think that nodes prepareStackTrace will get triggered and so being slower than reassigning the stacktrace from the original error to the aggregateError